### ISSUE TITLE
Test coverage GHA passing despite failing to calculate

### ISSUE THREAD

**Issue by aclark02-arcus** [pharmaR/riskassessment#800]
Latest commit on `master`:

https://github.com/pharmaR/riskassessment/actions/runs/9761008268/job/26941091990#step:8:18

_Originally posted by @aclark02-arcus in https://github.com/pharmaR/riskassessment/issues/656#issuecomment-2206255142_

**Comment by jthompson-arcus** [comment]
@dgkf do you by chance know how to resolve this issue? I saw that you did some work on `{covr}` so I wasn't sure if you had more insight. I noticed that `{riskmetric}` is having the same issue. I tried a different workflow with #807 but it fails with "Not valid tokenless upload" despite (to my knowledge) passing the token.

**Comment by dgkf** [comment]
It looks like it's not a covr issue, but a rate limit imposed by `codecov` using unauthenticated uploads

```
[1] "Rate limit reached. Please upload with the Codecov repository upload token to resolve issue. Expected time to availability: 1598s."
```

And it seems that there was a [change to the r-lib workflows about 2mo ago](https://github.com/r-lib/actions/commit/76c974b3ad2b2c7c5b222c65b28704919f9291be) that silently ignores codecov upload failures. Our workflows were most recently updated 7mo ago so we don't have these changes, but I think this would just suppress them, not solve them. 

You can either deal with the occasional rate-limiting, or add a `CODECOV_TOKEN` secret to the repo which will automatically be used to do authenticated uploads. 

If you log in through GitHub auth at https://app.codecov.io you should see this repo in the dropdown and you can find the token in the "configure" menu next to the repo name.

**Comment by jthompson-arcus** [comment]
Thanks for the input. I had found the information about `CODECOV_TOKEN` but wasn't putting all the pieces together. Looks like we needed to both update our process and utilize the token.

**Comment by aclark02-arcus** [comment]
closed with #796
